routes/extra_data_encoding.go

```diff
@@ -0,0 +1,200 @@
+	if encoding, exists := specialExtraDataKeysToEncoding[extraDataKey]; exists {
+		return encoding
+	}
+	return ExtraDataEncoding{Decode: DecodeString, Encode: EncodeString}
+}
+
+func ReservedFieldCannotOverride(_ string) ([]byte, error) {
+	return nil, errors.Errorf("Reserved extra data field. Client writes to this field not supported.")
+}
+
+func DecodeString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	return string(bytes)
+}
+
+func EncodeString(str string) ([]byte, error) {
+	return []byte(str), nil
+}
+
+// Decode64BitIntString supports decoding integers up to a length of 8 bytes
+func Decode64BitIntString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	var decoded, _ = lib.Varint(bytes)
+	return strconv.FormatInt(decoded, 10)
+}
+
+// Encode64BitIntString supports encoding integers up to a length of 8 bytes
+func Encode64BitIntString(str string) ([]byte, error) {
+	var encoded, err = strconv.ParseInt(str, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	var buffer []byte
+	lib.PutVarint(buffer, encoded)
+	return buffer, nil
+}
+
+// Decode64BitUintString supports decoding unsigned integers up to a length of 8 bytes
+func Decode64BitUintString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	var decoded, _ = lib.Uvarint(bytes)
+	return strconv.FormatUint(decoded, 10)
+}
+
+// Encode64BitUintString supports decoding unsigned integers up to a length of 8 bytes
+func Encode64BitUintString(str string) ([]byte, error) {
+	var encoded, err = strconv.ParseUint(str, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	var buffer []byte
+	lib.PutUvarint(buffer, encoded)
+	return buffer, nil
+}
+
+func DecodeBoolString(bytes []byte, params *lib.DeSoParams, utxoView *lib.UtxoView) string {
+	return Decode64BitUintString(bytes, params, utxoView)
+}
+
+func EncodeBoolString(str string) ([]byte, error) {
+	return Encode64BitUintString(str)
+}
+
+func DecodeHexString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	return hex.EncodeToString(bytes)
+}
+
+func EncodeHexString(str string) ([]byte, error) {
+	return hex.DecodeString(str)
+}
+
+func EncodePkStringToBytes(str string) ([]byte, error) {
+	result, _, err := lib.Base58CheckDecode(str)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func DecodePkToString(bytes []byte, params *lib.DeSoParams, _ *lib.UtxoView) string {
+	return lib.PkToString(bytes, params)
+}
+
+func DecodePubKeyToUint64MapString(bytes []byte, params *lib.DeSoParams, utxoView *lib.UtxoView) string {
+	var decoded, err = lib.DeserializePubKeyToUint64Map(bytes)
+	if err != nil {
+		glog.Errorf("Error marshaling public key to uint64 map to string: %v", err)
+		// Fall back to default []byte -> string casting
+		return DecodeString(bytes, params, utxoView)
+	}
+	mapWithDecodedKeys := map[string]uint64{}
+	for k, v := range decoded {
+		mapWithDecodedKeys[lib.PkToString(k.ToBytes(), params)] = v
+	}
+	return fmt.Sprint(mapWithDecodedKeys)
```

Comment:
What does this output look like?

Reply:
This decoder isn't changed by this PR. It only appears so because I renamed this file. I've changing the filename back to the original name so it's clearer what the PR changes